agent/provider/src/market/negotiator/builtin/demand_validation.rs

```diff
@@ -0,0 +1,144 @@
+use ya_agreement_utils::OfferDefinition;
+
+use crate::market::negotiator::factory::DemandValidationNegotiatorConfig;
+use crate::market::negotiator::{
+    AgreementResult, NegotiationResult, NegotiatorComponent, ProposalView,
+};
+
+/// Negotiator that verifies that all required fields are present in proposal.
+pub struct DemandValidation {
+    required_fields: Vec<String>,
+}
+
+impl DemandValidation {
+    pub fn new(config: &DemandValidationNegotiatorConfig) -> DemandValidation {
+        DemandValidation {
```

Comment:
It is really a type conversion, so we could use `From` here, but other negotiators do not do it as well so lets keep it as is.
Why `iter` + `map` + `collect` an not just
```
impl DemandValidation {
    pub fn new(config: &DemandValidationNegotiatorConfig) -> Self {
        let required_fields = config.required_fields.clone();
        Self { required_fields }
    }
}
```